proxy/source/markdown_parser.py

```diff
@@ -58,12 +58,18 @@ def _convert_crlf(input_str: str) -> str:
 
 
 def _parse_links(input_str: str) -> str:
-    return re.sub(
+    input_str = re.sub(
         r"\[([\w\W]+?)\]\(([\w\W]+?)\)",
         r'<a href="\2">\1</a>',
         input_str,
         flags=re.MULTILINE,
     )
+    return re.sub(
+        r"(?<!href=\")(https?:\/\/[-a-zA-Z0-9._~:/?#@!$&()*+,;=%]+')",
```

Comment:
```suggestion
        r"(?<!href=\")(https?:\/\/[-a-zA-Z0-9._~:/?#@!$&()*+,;=%]+)",
```

I think that dangling `'` may have been breaking the regex.